### DISCUSSION TITLE
Retention Policy/Settings

### DISCUSSION THREAD

**Discussion by yoharnu** [Zibbp/ganymede#128] [Ideas]
I recognize that this software is designed to archive, but I don't always want to archive them indefinitely. I think it would be cool if we could set a retention policy (either # of days or file size or both). Ideally, this would be on a per-channel basis. Further, being able to "lock" a VOD to prevent deletion might be nice, too. Like other tasks, this would be run on a schedule with an option to trigger it.

**Comment by Zibbp** [comment]
I originally said this was out of scope to previous issues asking the same thing but since so many people are requesting this feature I can look into adding it. I don't know when I'll get around to it.

**Reply by yoharnu** [reply]
After posting this, I saw a previous issue (#115) where you said it was out of scope and pointed to the API docs. I've already started working on a script to do what I need, but of course it would be better for me if it was eventually part of the app. But if it's not what you envisioned for your app, I respect that.

**Reply by Zibbp** [reply]
Feel free to post the script in a public repo if you're comfortable with that as I'm sure others would enjoy it.

**Reply by yoharnu** [reply]
For sure. I plan to make it OS agnostic and easily configurable for that reason.

**Reply by Zibbp** [reply]
Got around to implementing a first iteration of this. Available in [v1.2.12](https://github.com/Zibbp/ganymede/releases/tag/v1.2.12)
